docs/notebooks/burger.ipynb

```diff
@@ -0,0 +1,783 @@
+    "$$\\frac{\\mathrm{d}\\widehat{I}}{\\mathrm{d}u_0} = \\frac{\\partial I}{\\partial u} \\frac{\\mathrm{d}u}{\\mathrm{d}u_0}. \\tag{3}$$\n",
+    "Computing $\\partial J/\\partial u$ is straightforward since the functional (Eq. (1)) is written in terms of the solution variable $u$. In contrast, the Jacobian $\\mathrm{d}u/\\mathrm{d}u_0$ involves an expensive computational operation that is avoided by using the adjoint method.\n",
+    "\n",
+    "In summary, the adjoint system is defined by using the relation of the functional $I$ with the forward equation $F(u, u_0)$. Taking the total derivative of $F(u, u_0)$ with respect to $u_0$ yields a relationship for the solution Jacobian $\\mathrm{d}u/\\mathrm{d}u_0$:\n",
+    "$$\\frac{\\mathrm{d}}{\\mathrm{d}u_0} F(u, u_0) = \\frac{\\partial F(u, u_0)}{\\partial u} \\frac{\\mathrm{d}u}{\\mathrm{d}u_0} + \\frac{\\partial F(u, u_0)}{\\partial u_0} = 0 \\\\\n",
+    " \\implies \\frac{\\mathrm{d}u}{\\mathrm{d}u_0} = - \\left(\\frac{\\partial F(u, u_0)}{\\partial u}\\right)^{-1} \\frac{\\partial F(u, u_0)}{\\partial u_0}.$$\n",
+    "Substituting $\\mathrm{d}u/\\mathrm{d}u_0$ into Eq. (3), we have:\n",
+    "$$\\frac{\\mathrm{d}\\widehat{I}}{\\mathrm{d}u_0} = - \\frac{\\partial I}{\\partial u} \\left(\\frac{\\partial F(u, u_0)}{\\partial u}\\right)^{-1} \\frac{\\partial F(u, u_0)}{\\partial u_0}.$$\n",
+    "Taking the adjoint (Hermitian transpose) of the above equation:\n",
+    "$$\\frac{\\mathrm{d}\\widehat{I}}{\\mathrm{d}u_0}^* = - \\frac{\\partial F}{\\partial u_0}^* \\frac{\\partial F}{\\partial u}^{-*} \\frac{\\partial I}{\\partial u}^{*}$$\n",
+    "\n",
+    "Next, we define:\n",
+    "$$\\lambda = \\left(\\frac{\\partial F(u, u_0)}{\\partial u}\\right)^{-*} \\frac{\\partial I}{\\partial u}^* \n",
+    "  \\implies  \\left(\\frac{\\partial F(u, u_0)}{\\partial u}\\right)^{*} \\lambda = \\frac{\\partial I}{\\partial u}^*,$$\n",
+    "which is the adjoint system. Finally, we obtain the following expression for the adjoint-based gradient computation:\n",
+    "$$\\frac{\\mathrm{d}\\widehat{I}}{\\mathrm{d} u_0}^* = - \\frac{\\partial F}{\\partial u_0}^* \\lambda.$$\n",
+    "In this current case, Burger's equation is written in weak form and discretized in time with the backward finite difference method, we have:\n",
+    "$$ F(u^{n+1}, u^n, v) = \\int_{\\Omega} u^{n + 1} \\cdot v - u^{n}  \\cdot v + \\Delta t \\left(u^{n + 1} \\frac{\\partial u^{n + 1}}{\\partial x} \\cdot v + \\frac{\\partial u^{n + 1}}{\\partial x}  \\cdot \\frac{\\partial v}{\\partial x}\\right) \\, dx = 0 \\quad \\forall v\\in V \\tag{4},$$\n",
+    "where $v \\in V$ is an arbitrary test function.\n",
+    "\n",
+    "The time sequence of the adjoint system for the functional (Eq. (1)) is given by:\n",
+    "$$\n",
+    "\\begin{align*}\n",
+    "    \\lambda^{N+1} &= \\frac{\\partial I}{\\partial u^{N + 1}} \\\\\n",
```

Comment:
I am still thinking about how I should write this sequence of adjoint equations.